src/modules/Search/Search.js

```diff
@@ -552,6 +552,7 @@ export default class Search extends Component {
         key={childKey || result.title}
         active={selectedIndex === offsetIndex}
         onClick={this.handleItemClick}
+        onMouseDown={e => e.preventDefault()}
```

Comment:
1. Please move this code to a separate class method and place it under `handleItemClick`:

```jsx
  handleItemMouseDown = (e) => {
    debug('handleItemMouseDown()')
    // We should prevent default to prevent blur events.
    // https://github.com/Semantic-Org/Semantic-UI-React/issues/3298
    e.preventDefault()
  }
```

2. As we there, let's use `_.invoke()`:

```jsx
  handleFocus = (e) => {
    debug('handleFocus()')

    _.invoke(this.props, 'onFocus', e, this.props)
    this.setState({ focus: true })
  }

  handleBlur = (e) => {
    debug('handleBlur()')

    _.invoke(this.props, 'onBlur', e, this.props)
    this.setState({ focus: false })
  }
```

```diff
- onMouseDown={e => e.preventDefault()}
+ onMouseDown={this.handleItemMouseDown}
```

3. About a new unit test, we want to assert that `onBlur` will not happen on the item selection, we can test this:

```jsx
    it('is not called on an item click', () => {
      const onBlur = sandbox.spy()
      wrapperMount(<Search results={options} onBlur={onBlur} />)

      openSearchResults()
      wrapper.find('SearchResult').at('0').simulate('click', nativeEvent)
      onBlur.should.have.not.been.called()
    })
```